dds/DCPS/WriterInfo.cpp

```diff
@@ -239,28 +245,40 @@ WriterInfo::remove_instance(DDS::InstanceHandle_t instance)
   owner_evaluated_.erase(instance);
 }
 
-MonotonicTimePoint
-WriterInfo::check_activity(const MonotonicTimePoint& now)
+void
+WriterInfo::start_liveliness_timer()
 {
-  MonotonicTimePoint expires_at(MonotonicTimePoint::max_value);
-
   ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
+  last_liveliness_activity_time_ = MonotonicTimePoint::now();
+  if (reader_liveliness_lease_duration_is_finite_) {
+    liveliness_check_task_->schedule(reader_liveliness_lease_duration_);
+  }
+}
 
-  WriterInfoListener_rch reader = reader_.lock();
-
-  // We only need check the liveliness with the non-zero liveliness_lease_duration_.
-  if (state_ == ALIVE && reader && !reader->liveliness_lease_duration_.is_zero()) {
-    expires_at = last_liveliness_activity_time_ + reader->liveliness_lease_duration_;
-
-    if (expires_at <= now) {
-      // let all instances know this write is not alive.
-      guard.release();
-      reader->writer_became_dead(*this);
-      expires_at = MonotonicTimePoint::max_value;
+void
+WriterInfo::check_liveliness(const MonotonicTimePoint& now)
+{
+  bool became_dead;
+  WriterState previous_state;
+
+  {
+    ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
+    previous_state = state_;
+    if (state_ == ALIVE) {
+      const TimeDuration elapsed = now - last_liveliness_activity_time_;
+      became_dead = elapsed >= reader_liveliness_lease_duration_;
```

Comment:
I haven't followed all the code in this class, but a question I had is should the check use strictly greater comparison, i.e. `became_dead = elapsed > reader_liveliness_lease_duration_;`?

Reply:
I think using `>` makes more sense since it needs to exceed.